Orange/data/table/base.py

```diff
@@ -0,0 +1,1242 @@
+
+        The columns are in the same order as the columns in Table.domain.metas.
+
+        Returns
+        -------
+        np.ndarray
+        """
+        return self._to_numpy(meta=True)
+
+    @property
+    def weights(self):
+        """Get the weights as a 1D numpy array."""
+        if isinstance(self, SeriesBase):
+            # at least 1D when using this in TableSeries, which instead returns a 0D ndarray directly
+            result = np.atleast_1d(self[self._WEIGHTS_COLUMN])
+        else:
+            result = self[self._WEIGHTS_COLUMN].values
+        # even if weights are set with an integer, we need to return a float
+        return result.astype(float)
+
+    def set_weights(self, weight):
+        """Set the weights for the instances in this table.
+
+        Parameters
+        ----------
+        weight : Number or str or Sequence or np.ndarray or pd.Series
+            If a number, all weights are set to that value.
+            If a string, weights are set to whatever the column with that name's values are,
+            but only if those values are all numbers and are not NA/NaN.
+            If a sequence of (non-NA/NaN) numbers, set those values as the sequence.
+
+        Raises
+        ------
+        ValueError
+            If weights are nan, the column does not exist or the sequence length is mismatched.
+        TypeError
+            If an unrecognized type is passed.
+        """
+        if isinstance(weight, Number):
+            if np.isnan(weight):
+                raise ValueError("Weights cannot be nan. ")
+            new_weights = weight
+        elif isinstance(weight, str):
+            if weight not in self.columns:
+                raise ValueError("{} is not a column.".format(repr(weight)))
+            if self[weight].isnull().any() and np.issubdtype(self[weight].dtype, Number):
+                raise ValueError("All values in the target column must be valid numbers.")
+            new_weights = self[weight].fillna(value=self[weight].median())
+        elif isinstance(weight, Sequence):
+            if len(weight) != len(self):
+                raise ValueError("The sequence has length {}, expected length {}.".format(len(weight), len(self)))
+            new_weights = weight
+        elif isinstance(weight, np.ndarray):  # np.ndarray is not a Sequence
+            # allow row or column vectors
+            if weight.ndim > 1 and not (weight.ndim == 2 and weight.shape[1] == 1):
+                raise ValueError("Dimension mismatch.")
+            new_weights = np.ravel(weight)
+            if len(weight) != len(self):
+                raise ValueError("There are {} weights, expected {}.".format(len(weight), len(self)))
+        elif isinstance(weight, pd.Series):  # not only SeriesBase
+            # drop everything but the values to uncomplicate things
+            if weight.isnull().any():
+                raise ValueError("Weights cannot be nan. ")
```

Comment:
There is a bit of redundancy, but you lose some descriptive error messages. I don't think it's worth it just to save a few LOC.